### PR TITLE
Auto-configure region with InstanceProfileRegionProvider

### PR DESCRIPTION
**Pull request by mcruzdev**
## :loudspeaker: Type of change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring

## :scroll: Description

`RegionProviderAutoconfiguration` was changed to issue a `InstanceProfileRegionProvider` configured by `spring.cloud.aws.region.instance-profile`  property, similar to `CredentialsProviderAutoconfiguration`

## :bulb: Motivation and Context

#71 Fixes

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [ ] I updated reference documentation to reflect the change
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: v3/spring-cloud-aws-v3-autoconfigure/src/test/java/io/awspring/cloud/v3/autoconfigure/RegionProviderAutoConfigurationTests.java [hunk 1]
```diff
@@ -51,7 +54,7 @@ void autoDetectionConfigured_emptyStaticRegionConfigured_DefaultAwsRegionProvide
 	@Test
 	void staticRegionConfigured_staticRegionProviderWithConfiguredRegionConfigured() {
 		this.contextRunner.withPropertyValues("spring.cloud.aws.region.static:eu-west-1").run((context) -> {
-			StaticRegionProvider regionProvider = context.getBean(StaticRegionProvider.class);
+			AwsRegionProvider regionProvider = context.getBean(AwsRegionProvider.class);
```
Comment (maciejwalkowiak) [inline v3/spring-cloud-aws-v3-autoconfigure/src/test/java/io/awspring/cloud/v3/autoconfigure/RegionProviderAutoConfigurationTests.java:57]: We should also check here that it is an instance of static region provider.

**Review by maciejwalkowiak — CHANGES_REQUESTED** [review]
Thanks a lot for contribution! Just one small thing to change and we're getting it merged :-)

File: v3/spring-cloud-aws-v3-autoconfigure/src/test/java/io/awspring/cloud/v3/autoconfigure/RegionProviderAutoConfigurationTests.java (hunk 1 above)
Comment (mcruzdev) [reply v3/spring-cloud-aws-v3-autoconfigure/src/test/java/io/awspring/cloud/v3/autoconfigure/RegionProviderAutoConfigurationTests.java]: Thank you @maciejwalkowiak, solved ;)

**Review by maciejwalkowiak — APPROVED** [review]

**Comment by maciejwalkowiak** [comment]
Thanks a lot @mcruzdev for the contribution!

**Pull request merged**